smb_streams.c

```diff
@@ -286,6 +287,39 @@ static int php_smb_ops_seek(php_stream *stream, off_t offset, int whence, off_t
 	return -1;
 }
 
+static int php_smb_ops_set_option(php_stream *stream, int option, int value, void *ptrparam TSRMLS_DC)
+{
+	size_t newsize;
+	STREAM_DATA_FROM_STREAM();
+
+	if (!self || !self->handle) {
+		return PHP_STREAM_OPTION_RETURN_ERR;
+	}
+	if (!self->smbc_ftruncate) {
+		self->smbc_ftruncate = smbc_getFunctionFtruncate(self->state->ctx);
+	}
+	if (!self->smbc_ftruncate) {
+		return PHP_STREAM_OPTION_RETURN_ERR;
+	}
+
+	switch(option) {
+		case PHP_STREAM_OPTION_TRUNCATE_API:
+			switch (value) {
+				case PHP_STREAM_TRUNCATE_SUPPORTED:
```

Comment:
What about a default here?

Reply:
fallback to PHP_STREAM_OPTION_RETURN_NOTIMPL (btw, other default can be removed)

Reply:
done, default removed, and return become the last command of the function